### ISSUE TITLE
Securing Routes: add middleware example for resource routes

### ISSUE THREAD

**Issue by connor11528** [adonisjs/legacy-documentation#35]
I was looking at the [documentation on securing routes](http://adonisjs.com/docs/3.2/authentication#_securing_routes) and had a question.

I've defined routes using route resource. I would like to use the auth middleware to protect the creating, editing and deleting routes but not the showing and index routes.

How do I apply middleware to specific routes defined as a resource?

Example routes:

```
Route.resource('tasks', 'TaskController')
// .middleware('auth') <-- this breaks cause it hides the task index and show routes for unauthenticated users

Route.get('/login', 'AuthController.showLogin')
Route.post('/login', 'AuthController.login')

Route.get('/register', 'AuthController.showRegister')
Route.post('register', 'AuthController.register')

Route.get('/logout', 'AuthController.logout')
```

My task controller looks like this:

```
'use strict'

class TaskController {

  static get inject () { 
    return ['App/Model/Task']  
  }

  constructor (Task) { 
    this.Task = Task
  }

  * index(request, response) {
    const tasks = yield this.Task.all() 
    yield response.sendView('tasks.index', { tasks: tasks.toJSON() })
  }

  * create(request, response) {
    yield response.sendView('tasks.create')
  }

  * store(request, response) {
    let task = request.only('title', 'description');

    const newTask = new this.Task({
      title: task.title,
      description: task.description,
      user_id: request.currentUser
    })

    yield newTask.save()
  }

  * show(request, response) {
    const task = yield this.Task.find(request.param('id'))

    if (task) {
      yield response.sendView('tasks.show', { task: task.toJSON() })
      return
    }

    response.send('Sorry, cannot find the selected found')
  }

  * edit(request, response) {
    yield response.sendView('tasks.edit')
  }

  * update(request, response) {
    //
  }

  * destroy(request, response) {
    const task = yield this.Task.findBy('id', request.param('id'))
    yield task.delete()
  }

}

module.exports = TaskController
```

I'm fine protecting the routes using middleware in the route or controller definitions. What is the best way to protect the routes so that only authenticated users can create, edit and delete tasks?

**Comment by connor11528** [comment]
I suppose I could add 

```
const isLoggedIn = yield request.auth.check()

    if (!isLoggedIn) {
      response.redirect('/login')
    }
```

To every controller method but open to a cleaner way of doing it. Also, it'd probably be good to check that it updating or deleting is going down that the `user_id` for the task matches the authenticated user's id

Help greatly appreciated with this!

**Comment by RomainLanz** [comment]
Hey @connor11528! 👋 

I saw that you are using 3.2 but unfortunately, this feature is only available in 4.0.

**Comment by connor11528** [comment]
Okay and 4.0 is not released yet.

In Laravel we can do something like:

```
    public function __construct()
    {
        $this->middleware('auth')->except(['index', 'show']);
    }
```

Will something like this work in Adonis?

**Comment by RomainLanz** [comment]
Here's the doc about this feature in 4.0 http://dev.adonisjs.com/docs/4.0/routing#_resource_middleware

**Comment by connor11528** [comment]
Okay got it. Happy that it is included in 4.0 release!
